### PR TITLE
Add support to sig-proxy for podman-remote

### LINKED ISSUE #14707: Ctrl+C doesn't kill the container using `podman run -i` with podman machine
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

It's impossible to stop container attached to the terminal (flags `-a`, `-i` or `-t`) by pressing Ctrl+C. 

**Steps to reproduce the issue:**

1. `podmain machine start`

1. `podman run -i nginx`

2. press `ctrl+c`

3. `podman ps --noheading`

**Describe the results you received:**
```
330f2d2ee822  docker.io/library/nginx:latest  nginx -g daemon o...  14 hours ago  Up 14 hours ago              wonderful_kowalevski
```

**Describe the results you expected:**
```
(empty output)
```

**Additional information you deem important (e.g. issue happens only occasionally):**
`Created` and `Status` in `docker ps` output are suspicious. They say `14 hours` despite the fact that container is several seconds old.

```
➜  ~ podman run -i --log-level=debug nginx
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called run.PersistentPreRunE(podman run -i --log-level=debug nginx) 
DEBU[0000] SSH Ident Key "/Users/ponimas/.ssh/podman-machine-default" SHA256:tOvATDNZzT3jTcmqxc4SrKQnIicNP17+MTCNpfV3FIk ssh-ed25519 
DEBU[0000] Found SSH_AUTH_SOCK "/var/folders/ff/y12n_58x3yq88ysv7jk864p40000gn/T//ssh-itLGKjzLvN9B/agent.595", ssh-agent signer(s) enabled 
DEBU[0000] SSH Agent Key SHA256:ucuqyTCiUBQSM+huZ1uxzQHtO9OMdHFwrvuUlxwXxMM ssh-rsa 
DEBU[0000] DoRequest Method: GET URI: http://d/v4.1.1/libpod/_ping 
DEBU[0000] Loading registries configuration "/etc/containers/registries.conf" 
DEBU[0000] No credentials matching pierone.stups.zalan.do found in /Users/ponimas/.config/containers/auth.json 
DEBU[0000] No credentials matching pierone.stups.zalan.do found in /Users/ponimas/.config/containers/auth.json 
DEBU[0000] Looking up in credential helper pierone based on credHelpers entry in /Users/ponimas/.docker/config.json 
DEBU[0000] Found credentials for pierone.stups.zalan.do in credential helper containers-auth.json in file /Users/ponimas/.docker/config.json 
DEBU[0000] DoRequest Method: POST URI: http://d/v4.1.1/libpod/images/pull 
DEBU[0000] DoRequest Method: POST URI: http://d/v4.1.1/libpod/containers/create 
INFO[0000] Going to attach to container "ca8d4efa8af12f0fe13a9162dd5757d79cfb6196aced5e4da4eee8a36ce48185" 
DEBU[0000] DoRequest Method: GET URI: http://d/v4.1.1/libpod/containers/ca8d4efa8af12f0fe13a9162dd5757d79cfb6196aced5e4da4eee8a36ce48185/json 
DEBU[0000] DoRequest Method: POST URI: http://d/v4.1.1/libpod/containers/ca8d4efa8af12f0fe13a9162dd5757d79cfb6196aced5e4da4eee8a36ce48185/attach 
DEBU[0000] Copying standard streams of container "ca8d4efa8af12f0fe13a9162dd5757d79cfb6196aced5e4da4eee8a36ce48185" in non-terminal mode 
DEBU[0000] Copying STDIN to socket                      
INFO[0000] Going to start container "ca8d4efa8af12f0fe13a9162dd5757d79cfb6196aced5e4da4eee8a36ce48185" 
DEBU[0000] DoRequest Method: POST URI: http://d/v4.1.1/libpod/containers/ca8d4efa8af12f0fe13a9162dd5757d79cfb6196aced5e4da4eee8a36ce48185/start 
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: Getting the checksum of /etc/nginx/conf.d/default.conf
10-listen-on-ipv6-by-default.sh: info: Enabled listen on IPv6 in /etc/nginx/conf.d/default.conf
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/06/22 15:14:23 [notice] 1#1: using the "epoll" event method
2022/06/22 15:14:23 [notice] 1#1: nginx/1.21.6
2022/06/22 15:14:23 [notice] 1#1: built by gcc 10.2.1 20210110 (Debian 10.2.1-6) 
2022/06/22 15:14:23 [notice] 1#1: OS: Linux 5.18.5-200.fc36.aarch64
2022/06/22 15:14:23 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 524288:524288
2022/06/22 15:14:23 [notice] 1#1: start worker processes
2022/06/22 15:14:23 [notice] 1#1: start worker process 26
2022/06/22 15:14:23 [notice] 1#1: start worker process 27
^C
➜  ~ 
```

**Output of `podman version`:**

```
lient:       Podman Engine
Version:      4.1.1
API Version:  4.1.1
Go Version:   go1.18.3
Built:        Tue Jun 14 22:12:46 2022
OS/Arch:      darwin/arm64

Server:       Podman Engine
Version:      4.1.0
API Version:  4.1.0
Go Version:   go1.18.2
Built:        Mon May 30 18:04:32 2022
OS/Arch:      linux/arm64
```

**Output of `podman info --debug`:**

```
host:
  arch: arm64
  buildahVersion: 1.26.1
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.0-2.fc36.aarch64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.0, commit: '
  cpuUtilization:
    idlePercent: 99.7
    systemPercent: 0.2
    userPercent: 0.1
  cpus: 2
  distribution:
    distribution: fedora
    variant: coreos
    version: "36"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
    uidmap:
    - container_id: 0
      host_id: 501
      size: 1
    - container_id: 1
      host_id: 100000
      size: 1000000
  kernel: 5.18.5-200.fc36.aarch64
  linkmode: dynamic
  logDriver: journald
  memFree: 3513565184
  memTotal: 4090519552
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.4.5-1.fc36.aarch64
    path: /usr/bin/crun
    version: |-
      crun version 1.4.5
      commit: c381048530aa750495cf502ddb7181f2ded5b400
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/501/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.aarch64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 0
  swapTotal: 0
  uptime: 4h 22m 12.55s (Approximately 0.17 days)
plugins:
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  volume:
  - local
registries:
  search:
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/home/core/.local/share/containers/storage
  graphRootAllocated: 106825756672
  graphRootUsed: 3219861504
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 4
  runRoot: /run/user/501/containers
  volumePath: /var/home/core/.local/share/containers/storage/volumes
version:
  APIVersion: 4.1.0
  Built: 1653926672
  BuiltTime: Mon May 30 18:04:32 2022
  GitCommit: ""
  GoVersion: go1.18.2
  Os: linux
  OsArch: linux/arm64
  Version: 4.1.0
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
brew info podman
podman: stable 4.1.1 (bottled), HEAD
Tool for managing OCI containers and pods
https://podman.io/
/opt/homebrew/Cellar/podman/4.1.1 (174 files, 46.4MB) *
  Poured from bottle on 2022-06-16 at 06:53:11
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/podman.rb
License: Apache-2.0
==> Dependencies
Build: go ✘, go-md2man ✘
Required: qemu ✘
==> Options
--HEAD
	Install HEAD version
==> Caveats
zsh completions have been installed to:
  /opt/homebrew/share/zsh/site-functions
==> Analytics
install: 19,991 (30 days), 60,518 (90 days), 161,801 (365 days)
install-on-request: 19,812 (30 days), 60,294 (90 days), 161,556 (365 days)
build-error: 4 (30 days)
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

Yes/~No~

**Additional environment details (AWS, VirtualBox, physical, etc.):**

```
OSX Monterey 12.4
MacBook Pro (14-inch, 2021)
Chip:	Apple M1 Max
```

### PR DESCRIPTION
**Pull request by boaz0**
closes #14707 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go [hunk 1]
```diff
@@ -16,6 +16,48 @@ import (
 // runc is using the same value.
 const signalBufferSize = 2048
 
+func ProxySignalsRemote(ctrID string, killFunc func() error) {
```
Comment (boaz0) [inline pkg/domain/infra/abi/terminal/sigproxy_linux.go:19]: I just copy paste `ProxySignals` to a new method that accepts a kill function. I can refactor this and use it also in non-remote parts as well. Just lemme know what you think @mheon

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go [hunk 2]
```diff
@@ -16,6 +16,48 @@ import (
...
+	// Stop catching the shutdown signals (SIGINT, SIGTERM) - they're going
+	// to the container now.
+	shutdown.Stop() //nolint: errcheck
```
Comment (mheon) [inline pkg/domain/infra/abi/terminal/sigproxy_linux.go:22]: I don't know if you need this bit, the shutdown handler should only be invoked by non-Remote Podman

File: pkg/domain/infra/tunnel/containers.go [hunk 3]
```diff
@@ -819,6 +820,14 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 	}
 
 	// Attach
+	if opts.SigProxy {
+		terminal.ProxySignalsRemote(con.ID, func() error {
+			killOpts := entities.KillOptions{All: false, Latest: false, Signal: "TERM"}
```
Comment (mheon) [inline pkg/domain/infra/tunnel/containers.go:825]: This isn't correct, we don't always proxy TERM. You need to proxy the signal that was received.

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go (hunk 2 above)
Comment (boaz0) [reply pkg/domain/infra/abi/terminal/sigproxy_linux.go]: No problem. I removed that line.

File: pkg/domain/infra/tunnel/containers.go (hunk 3 above)
Comment (boaz0) [reply pkg/domain/infra/tunnel/containers.go]: Yeah, I eventually had to pass the `syscall.Signal` and convert it to string. I am pretty sure there is a better way to convert it to string so let me know what I'm missing.

Thanks.

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go [hunk 4]
```diff
@@ -16,6 +16,44 @@ import (
 // runc is using the same value.
 const signalBufferSize = 2048
 
+func ProxySignalsRemote(ctrID string, killFunc func(sig syscall.Signal) error) {
```
Comment (boaz0) [inline pkg/domain/infra/abi/terminal/sigproxy_linux.go:19]: I have no idea why this one is undefined on MacOS... @mheon any ideas?
Thanks.

Comment (Luap99) [reply pkg/domain/infra/abi/terminal/sigproxy_linux.go:19]: Because the file is named `_linux.go` therefore it will only compile on linux.

I recommend to move this function somewhere else, maybe `pkg/domain/infra/tunnel`? `pkg/domain/infra/abi` should only be used by local podman code and not the remote client.

File: pkg/domain/infra/tunnel/containers.go [hunk 5]
```diff
@@ -819,6 +821,14 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 	}
 
 	// Attach
+	if opts.SigProxy {
+		terminal.ProxySignalsRemote(con.ID, func(sig syscall.Signal) error {
+			killOpts := entities.KillOptions{All: false, Latest: false, Signal: strconv.FormatInt(int64(sig), 10)}
```
Comment (Luap99) [inline pkg/domain/infra/tunnel/containers.go:826]: sending a number in not safe, across different operating systems and architectures the numbers can vary.
We must should send the actual signal name.

I recommend to look at pkg/signal, you should add a new function there which loop over the SIgnalMap and converts the syscall.Signal into a the actual SIGNAL name.

File: go.mod [hunk 6]
```diff
@@ -3,7 +3,7 @@ module github.com/containers/podman/v4
 go 1.16
 
 require (
-	github.com/BurntSushi/toml v1.1.0
+	github.com/BurntSushi/toml v1.2.0
```
Comment (Luap99) [inline go.mod:6]: This is not related to this PR please drop the commit.

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go (hunk 4 above)
Comment (Luap99) [reply pkg/domain/infra/abi/terminal/sigproxy_linux.go:19]: Yes that is correct, although we call podman-remote just podman on macos and windows.

File: go.mod (hunk 6 above)
Comment (boaz0) [reply go.mod]: Fixed. Thanks.

File: pkg/domain/infra/tunnel/containers.go (hunk 5 above)
Comment (boaz0) [reply pkg/domain/infra/tunnel/containers.go]: Done. Thanks.

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go [hunk 7]
```diff
@@ -16,6 +16,44 @@ import (
...
+			// intended for the podman command itself.
+			// SIGURG was added because of golang 1.14 and its preemptive changes
+			// causing more signals to "show up".
+			// https://github.com/containers/podman/issues/5483
+			if s == syscall.SIGCHLD || s == syscall.SIGPIPE || s == syscall.SIGURG {
+				continue
+			}
+
+			if err := killFunc(s.(syscall.Signal)); err != nil {
+				if errors.Is(err, define.ErrCtrStateInvalid) {
```
[the remark this answers is not in the document]
Comment (boaz0) [reply pkg/domain/infra/abi/terminal/sigproxy_linux.go]: I am not sure I understood you. Can you make sure I fixed it correctly here:
https://github.com/containers/podman/pull/15131/files#diff-17b8a2d11834d1c343b405ad3b77c559d5fb3f856c6b4178c74b3c54e063d746R51
Thanks

File: pkg/domain/infra/abi/terminal/sigproxy_linux.go (hunk 4 above)
Comment (boaz0) [reply pkg/domain/infra/abi/terminal/sigproxy_linux.go]: Thanks for your clarification!

**Comment by TomSweeneyRedHat** [comment]
Quick review and the code LGTM, but the tests aren't a bit happy.

File: pkg/domain/infra/tunnel/runtime.go [hunk 8]
```diff
@@ -18,3 +25,45 @@ type ContainerEngine struct {
...
+				} else {
+					logrus.Errorf("forwarding signal %d to container %s: %v", s, ctrID, err)
+				}
+				// If the container dies, and we find out here,
+				// we need to forward that one signal to
+				// ourselves so that it is not lost, and then
+				// we terminate the proxy and let the defaults
+				// play out.
+				signal.StopCatch(sigBuffer)
+				if err := syscall.Kill(syscall.Getpid(), s.(syscall.Signal)); err != nil {
```
Comment (rhatdan) [inline pkg/domain/infra/tunnel/runtime.go]: Doesn't this have to happen on the server side?

Comment (boaz0) [reply pkg/domain/infra/tunnel/runtime.go]: I don't think so. @mheon am I right?

**Comment by boaz0** [comment]
@edsantiago do you have an idea how should I test this. Should I run `podman run -i nginx` in the background and run `kill -9 <pid>`?

I saw there are some tests in https://github.com/containers/podman/blob/main/test/e2e/run_signal_test.go but I am not sure if they are related to the issue. In addition, when I run them on my local environment they fail.

//cc @TomSweeneyRedHat

**Comment by edsantiago** [comment]
The following works (says BYE) with your PR, does not work (silence) on main:
```console
$ bin/podman-remote run --name foo $iii sh -c 'trap "echo BYE;exit 0" INT;while :;do sleep 0.1;done' &
...
$ kill -INT %1
...
$ podman logs foo
BYE
```
Making that into a proper test -- capturing the PID (instead of %1), using randomized strings, and cleaning up -- is left as an exercise for the reader. HTH.

**Comment by boaz0** [comment]
@edsantiago I tried for too long to make this work in the system tests but it keeps failing. When I do it manually it works but running it through `bats` doesn't show the same results.

**Comment by boaz0** [comment]
@edsantiago you're amazing. I have no idea how you figured it out but it works.

**Comment by edsantiago** [comment]
@boaz0 blush, all it is is familiarity with one obscure little fraction of code. Still, thank you!

@containers/podman-maintainers PTAL; the new imports are triggering bloat checks and Mac/Windows build failures. There might be a reason why those modules aren't imported in podman-remote :-(

**Comment by mheon** [comment]
Code changes LGTM. Not sure what's causing the bloat... My assumption would be the terminal package, but I don't see what in there could be so big

**Review by vrothberg — COMMENTED** [review]
LGTM
I re-triggered the failed jobs

**Comment by vrothberg** [comment]
The cross-build failures look legit: https://github.com/containers/podman/pull/15131/checks?check_run_id=8214190961

@boaz0 can you take another look?

**Review by Luap99 — CHANGES_REQUESTED** [review]
Please do not ignore the bloat check here. There is something very wrong!
This PR causes a import of `github.com/containers/podman/v4/libpod` on the remote client. This should never happen.

Generally speaking, I don't we should ever import anything from `pkg/domain/infra/abi` in `pkg/domain/infra/tunnel`. This will just cause headaches later on when trying to find out why the remote client is so bloated.

I strongly recommend to move the terminal package somewhere else and you have to remove the `github.com/containers/podman/v4/libpod` dependency

**Comment by boaz0** [comment]
OK, I am going to move the `terminal` part to a different package and hopefully this will fix the `bloat` issue.

File: pkg/domain/infra/tunnel/runtime.go [hunk 9]
```diff
@@ -18,3 +24,27 @@ type ContainerEngine struct {
...
+
+	go func() {
+		for s := range sigBuffer {
+			signalName, err := signal.ParseSysSignalToName(s.(syscall.Signal))
+			if err != nil {
+				logrus.Infof("Ceasing signal forwarding to container %s as it has stopped", ctrID)
+			}
+			if err := killFunc(signalName); err != nil {
+				if err.Error() == define.ErrCtrStateInvalid.Error() {
+					logrus.Infof("Ceasing signal forwarding to container %s as it has stopped", ctrID)
```
Comment (rhatdan) [inline pkg/domain/infra/tunnel/runtime.go:43]: Should probably be logrus.Debugf

File: test/system/032-sig-proxy.bats [hunk 10]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bats
+
+load helpers
+
+@test "podman sigkill" {
+    $PODMAN run -i --name foo $IMAGE sh -c 'trap "echo BYE;exit 0" INT;echo READY;while :;do sleep 0.1;done' &
```
Comment (rhatdan) [inline test/system/032-sig-proxy.bats:6]: Why not run this container as a daemon?

File: test/system/032-sig-proxy.bats [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    local kidpid=$!
+
+    # Wait for container to appear
```
Comment (rhatdan) [inline test/system/032-sig-proxy.bats:9]: I don't think this loop is necessary.  The container should be there instantly if you run it with a -d flag.

File: pkg/domain/infra/tunnel/runtime.go [hunk 12]
```diff
@@ -18,3 +24,27 @@ type ContainerEngine struct {
...
+				} else {
+					logrus.Errorf("forwarding signal %d to container %s: %v", s, ctrID, err)
```
[the remark this answers is not in the document]
Comment (boaz0) [reply pkg/domain/infra/tunnel/runtime.go]: Fixed that. I wrongly copy-pasted this `if-else` statement.

File: pkg/domain/infra/tunnel/runtime.go [hunk 13]
```diff
@@ -18,3 +24,27 @@ type ContainerEngine struct {
...
+	sigBuffer := make(chan os.Signal, signal.SignalBufferSize)
+	signal.CatchAll(sigBuffer)
+
+	logrus.Debugf("Enabling signal proxying")
+
+	go func() {
+		for s := range sigBuffer {
+			signalName, err := signal.ParseSysSignalToName(s.(syscall.Signal))
+			if err != nil {
+				logrus.Infof("Ceasing signal forwarding to container %s as it has stopped", ctrID)
```
[the remark this answers is not in the document]
Comment (boaz0) [reply pkg/domain/infra/tunnel/runtime.go]: done

File: pkg/domain/infra/tunnel/runtime.go (hunk 9 above)
Comment (boaz0) [reply pkg/domain/infra/tunnel/runtime.go:43]: done

File: pkg/domain/infra/tunnel/runtime.go [hunk 14]
```diff
@@ -18,3 +24,25 @@ type ContainerEngine struct {
 type SystemEngine struct {
 	ClientCtx context.Context
 }
+
+// RemoteProxySignals enables sig-proxy in remote mode.
+func RemoteProxySignals(ctrID string, killFunc func(string) error) {
```
Comment (vrothberg) [inline pkg/domain/infra/tunnel/runtime.go:29]: I think this can be a private function as it's exclusively used in the `tunnel` package.

File: pkg/domain/infra/tunnel/runtime.go (hunk 9 above)
Comment (vrothberg) [reply pkg/domain/infra/tunnel/runtime.go:43]: It's still `Infof`.

File: pkg/signal/signal_common.go [hunk 15]
```diff
@@ -56,3 +60,12 @@ func StopCatch(sigc chan os.Signal) {
 	signal.Stop(sigc)
 	close(sigc)
 }
+
+func ParseSysSignalToName(s syscall.Signal) (string, error) {
```
Comment (vrothberg) [inline pkg/signal/signal_common.go:64]: Needs a doc comment.

File: pkg/signal/signal_common.go [hunk 16]
```diff
@@ -56,3 +60,12 @@ func StopCatch(sigc chan os.Signal) {
...
+	for k, v := range SignalMap {
+		if v == s {
+			return k, nil
+		}
+	}
+	return "", fmt.Errorf("invalid syscall signal: %s", s)
```
Comment (vrothberg) [inline pkg/signal/signal_common.go:70]:
```suggestion
	return "", fmt.Errorf("unknown syscall signal: %v", s)
```

**Review by vrothberg — COMMENTED** [review]
Just minor final nits.
Thanks for working on this, @boaz0 !

File: pkg/signal/signal_common.go (hunk 16 above)
Comment (boaz0) [reply pkg/signal/signal_common.go]: done

File: pkg/signal/signal_common.go (hunk 15 above)
Comment (boaz0) [reply pkg/signal/signal_common.go:66]: done

File: pkg/domain/infra/tunnel/runtime.go (hunk 14 above)
Comment (boaz0) [reply pkg/domain/infra/tunnel/runtime.go]: done

**Review by vrothberg — COMMENTED** [review]
Thanks, @boaz0!
LGTM
@Luap99 @rhatdan PTAL

**Review by Luap99 — APPROVED** [review]
LGTM

**Comment by edsantiago** [comment]
I restarted one flake, but am now seeing an alarming number of other test failures: [ubuntu](https://api.cirrus-ci.com/v1/artifact/task/6672156414181376/html/sys-remote-ubuntu-2204-root-host.log.html), [fedora](https://api.cirrus-ci.com/v1/artifact/task/4631462833029120/html/sys-remote-fedora-36-aarch64-root-host.log.html), both remote, both (mostly) hangs. I'm not going to restart those: there's something real-looking about them.

**Comment by boaz0** [comment]
@edsantiago - yep that doesn't look good. I am first rebasing and will try to reproduce these on my local environment.

**Comment by boaz0** [comment]
OK I think I know what's the problem. Let's see if that fixes the problem.
Thanks for the help @edsantiago

**Comment by boaz0** [comment]
@edsantiago looks like it fixed the problem but now `Windows Cross` failed. :thinking: I have a better idea.

**Comment by edsantiago** [comment]
@boaz0 thank you for your perseverance on this tricky one!
